backend/src/layer/data/swka_parser/html_parser.rs

```diff
@@ -369,20 +380,132 @@ impl HTMLParser {
             .collect()
     }
 
-    fn get_dish_type(dish_node: &ElementRef) -> Option<MealType> {
+    fn get_dish_type(dish_node: &ElementRef) -> Option<FoodType> {
         let dish_type_node = dish_node.select(&DISH_TYPE_NODE_CLASS_SELECTOR).next()?;
         dish_type_node
             .value()
             .attr(DISH_TYPE_ATTRIBUTE_NAME)
-            .map(MealType::parse)
+            .map(FoodType::parse)
     }
 
-    fn get_dish_env_score(dish_node: &ElementRef) -> Option<u32> {
-        let env_score_node = dish_node.select(&ENV_SCORE_NODE_CLASS_SELECTOR).next()?;
-        env_score_node
-            .value()
-            .attr(ENV_SCORE_ATTRIBUTE_NAME)?
-            .parse::<u32>()
+    fn get_dish_env_score(dish_node: &ElementRef) -> Option<EnvironmentInfo> {
+        let env_info = EnvironmentInfo {
+            average_rating: Self::get_average_env_score(dish_node)?,
+            co2_rating: Self::get_co2_rating(dish_node)?,
+            co2_value: Self::get_co2_value(dish_node)?,
+            water_rating: Self::get_rating(dish_node, "div.wasser_bewertung")?,
+            water_value: Self::get_water_value(dish_node)?,
+            animal_welfare_rating: Self::get_rating(dish_node, "div.tierwohl")?,
+            rainforest_rating: Self::get_rating(dish_node, "div.regenwald")?,
+            max_rating: Self::get_max_rating(dish_node, "div.regenwald")?,
+        };
+        Some(env_info)
+    }
+
+    fn get_average_env_score(dish_node: &ElementRef) -> Option<u32> {
+        Some(
```

Comment:
if this average is always calculated like that (manually) then we should probably not store it in the database in the first place and rather calculate it on access

Reply:
I think this could be best done on database access